### PR TITLE
Candidate REST API tests.

### PR DESCRIPTION
**Pull request by sittipat-chandang**
Solves #25 
Wrote a number of tests for the candidate API, and also make some error responses send error status code when it previously didn't. Please review, thank you.

### CODE REVIEW

File: apis/tests.py [hunk 1]
```diff
@@ -166,3 +168,286 @@ def test_vote_without_area(self):
...
+            "description": description,
+            "area_id": test_area_id
+        })
+        response_content = json.loads(response.content.decode("utf-8"))
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(response_content["result"]["user"]["id"], test_user_id)
+        self.assertEqual(response_content["result"]["description"], description)
+        self.assertEqual(response_content["result"]["area"]["id"], test_area_id)
+
```
Comment (pontakornth) [inline apis/tests.py:230]: Please check the existence of new candidate too.

**Review by pontakornth — CHANGES_REQUESTED** [review]
Other than response, checking if the candidate is actually created is important.

Just check the `POST` valid one should be enough.

File: apis/tests.py (hunk 1 above)
Comment (sittipat-chandang) [reply apis/tests.py:230]: Alright I've added an existence check via id. Does it look okay?

File: apis/tests.py [hunk 2]
```diff
@@ -166,3 +169,294 @@ def test_vote_without_area(self):
...
+        # Check created candidate's existence.
+        candidate_exists = True
+        try:
+            NewCandidate.objects.get(id=response_content["result"]["id"])
+        except ObjectDoesNotExist:
+            candidate_exists = False
+        self.assertTrue(candidate_exists)
```
Comment (pontakornth) [inline apis/tests.py]: Django has `exists()` that check existence. The whole block can be reduced to.

```suggestion
self.assertTrue(NewCandidate.objects.filter(id=response_content["result"]["id"]).exists())
```

**Review by pontakornth — CHANGES_REQUESTED** [review]

File: apis/tests.py (hunk 2 above)
Comment (sittipat-chandang) [reply apis/tests.py]: Ah I see, I've modified my test as suggested.

**Review by pontakornth — APPROVED** [review]

**Review by HelloYeew — APPROVED** [review]

**Pull request merged**